dhis-2/dhis-api/src/main/java/org/hisp/dhis/scheduling/JobConfiguration.java

```diff
@@ -364,141 +289,89 @@ public JobParameters getJobParameters() {
+        ? null
+        : DurationFormatUtils.formatDurationHMS(lastFinished.getTime() - lastExecuted.getTime());
   }
 
-  public void setLastRuntimeExecution(String lastRuntimeExecution) {
-    this.lastRuntimeExecution = lastRuntimeExecution;
-  }
-
-  @JacksonXmlProperty
+  /** Kept for backwards compatibility of the REST API */
   @JsonProperty(access = JsonProperty.Access.READ_ONLY)
   public boolean isLeaderOnlyJob() {
-    return !jobType.isRunOnAllNodes();
-  }
-
-  public boolean isInMemoryJob() {
-    return inMemoryJob;
-  }
-
-  public void setInMemoryJob(boolean inMemoryJob) {
-    this.inMemoryJob = inMemoryJob;
+    return true;
   }
 
   /**
-   * Kept for backwards compatibility
+   * Kept for backwards compatibility of the REST API
    *
    * @see #getExecutedBy()
    */
-  @JacksonXmlProperty
   @JsonProperty(access = JsonProperty.Access.READ_ONLY)
   public String getUserUid() {
     return executedBy;
   }
 
-  @JsonProperty
-  public String getExecutedBy() {
-    return executedBy;
+  public String getQueueIdentifier() {
+    return getQueueName() == null ? getUid() : getQueueName();
   }
 
-  public void setExecutedBy(String executedBy) {
-    this.executedBy = executedBy;
+  /**
+   * @return true if this configuration is part of a queue, false otherwise
+   */
+  public boolean isUsedInQueue() {
+    return getQueueName() != null;
   }
 
-  public String getQueueIdentifier() {
-    return getQueueName() == null ? getUid() : getQueueName();
+  public boolean isRunOnce() {
+    return cronExpression == null && delay == null && queueName == null;
   }
 
-  public String getQueueName() {
-    return queueName;
+  /**
+   * @return the next time this job should run based on the {@link #getLastExecuted()} time
+   */
+  public Instant nextExecutionTime(Instant now) {
+    // for good measure we offset the last time by 1 second
+    Instant since = lastExecuted == null ? now : lastExecuted.toInstant().plusSeconds(1);
+    if (isUsedInQueue() && getQueuePosition() > 0) return null;
+    return switch (getSchedulingType()) {
+      case ONCE_ASAP -> nextOnceExecutionTime(since);
+      case FIXED_DELAY -> nextDelayExecutionTime(since);
+      case CRON -> nextCronExecutionTime(since);
+    };
   }
 
-  public void setQueueName(String name) {
-    this.queueName = name;
+  private Instant nextCronExecutionTime(@Nonnull Instant since) {
+    if (isUndefinedCronExpression(cronExpression)) return null;
+    SimpleTriggerContext context =
+        new SimpleTriggerContext(Clock.fixed(since, ZoneId.systemDefault()));
+    Date next = new CronTrigger(cronExpression).nextExecutionTime(context);
+    return next == null ? null : next.toInstant();
   }
 
-  public Integer getQueuePosition() {
-    return queuePosition;
+  private Instant nextDelayExecutionTime(@Nonnull Instant since) {
+    if (delay == null || delay <= 0) return null;
+    // always want to run delay after last start, right ways when never started
```

Comment:
this section of the comment `right ways when never started` doesn't read well & I don't know what it's supposed to mean.
Should it say `right away when never started` ?

Reply:
Oh yes, that is a typo that you guessed correct.